fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java

```diff
@@ -253,7 +255,25 @@ public ScalarOperator visitSlot(SlotRef node, Context context) {
                     resolvedField.getScope().getRelationId().equals(expressionMapping.getOuterScopeRelationId())) {
                 correlation.add(columnRefOperator);
             }
-            return columnRefOperator;
+
+            // If origin type is struct type, means that node contains subfield access
+            if (node.getTrueOriginType().isStructType()) {
+                Preconditions.checkArgument(node.getUsedStructFieldPos() != null, "StructType SlotRef must have" +
+                        "an non-empty usedStructFiledPos!");
+                Preconditions.checkArgument(node.getUsedStructFieldPos().size() > 0);
+                List<Integer> usedStructFieldPos = node.getUsedStructFieldPos();
```

Comment:
usedStructFieldPos looks like have same function with MapType, can we unify it?

Reply:
That's two different functions, I know it looks the same.